packages/expo-sensors/src/utils/isSensorEnabledAsync.web.ts

```diff
@@ -0,0 +1,64 @@
+type SensorEventName = 'deviceorientation' | 'devicemotion';
+
+// iOS 12.2 disables DeviceMotion by default now
+// https://github.com/w3c/deviceorientation/issues/57
+export async function guardSensorEventEnabledAsync(
+  eventName: SensorEventName,
+  timeout?: number
+): Promise<boolean> {
+  if (!isIOS()) {
+    return true;
+  }
+
+  try {
+    return await guardEventEnabledAsync(eventName, timeout);
+  } catch ({ message }) {
+    throw new Error(
+      message +
+        '\nEnable device orientation in Settings > Safari > Motion & Orientation Access' +
+        '\nalso ensure that you are hosting with https as DeviceMotion is now a secure API on iOS Safari.'
+    );
+  }
+}
+
+// throw error if the sensor is disabled.
+export async function guardEventEnabledAsync(
+  eventName: string,
+  timeout: number = 500
+): Promise<boolean> {
+  return new Promise((resolve, reject) => {
+    const id = setTimeout(() => {
+      window.removeEventListener(eventName, listener);
+      reject(new Error(`Cannot observe event: ${eventName}.`));
```

Comment:
This could help someone track down which event caused the error. If they aren't familiar with this permission and have a bunch of features it would help to see that deviceorientation or devicemotion were the problem.

Reply:
The only callers of this function are inside this file. One of them suppresses the error and the other explains what to do in Safari to enable the event anyway so there should be enough info surfaced to the caller.